src/result_processor.c

```diff
@@ -1590,3 +1592,230 @@ ResultProcessor *RPDepleter_New(StrongRef sync_ref) {
   ret->sync_ref = sync_ref;
   return &ret->base;
 }
+
+// Container for search results with scores from multiple upstream processors
+typedef struct {
+  SearchResult *searchResult; // The actual search result document with metadata
+  double *scores;             // Dynamic array of scores from each upstream processor
+  bool *hasScores;            // Dynamic array of flags indicating if each score is available
+  size_t numSources;          // Number of upstream processors (array size)
+} HybridSearchResult;
+
+// Constructor for HybridSearchResult
+static HybridSearchResult* HybridSearchResult_New(SearchResult *searchResult, size_t numUpstreams) {
+  HybridSearchResult* hybridResult = rm_calloc(1, sizeof(HybridSearchResult));
+  hybridResult->searchResult = searchResult;
+  hybridResult->numSources = numUpstreams;
+  hybridResult->scores = rm_calloc(numUpstreams, sizeof(double));
+  hybridResult->hasScores = rm_calloc(numUpstreams, sizeof(bool));
```

Comment:
Since we know the length of this array in advance and each upstream already has its index, you can use remove `hasScores` and just use `scores` with some initial value that cannot be a valid score, e.g., `-1`.
`hasScores` is fine as well just not necessary, FYI